styx-common/src/main/java/com/spotify/styx/util/BasicWorkflowValidator.java

```diff
@@ -149,45 +147,33 @@ private static boolean validateServiceAccount(String serviceAccount) {
     return matcher.matches();
   }
 
-  private void validateDockerExecConf(List<String> errors, WorkflowConfiguration config) {
-    validateDockerParam(errors, "dockerImage", config.dockerImage(),
-        config.dockerExecConf().flatMap(DockerExecConf::dockerImage),
-        this::validateDockerImage);
-    validateDockerParam(errors, "dockerArgs", config.dockerArgs(),
-        config.dockerExecConf().flatMap(DockerExecConf::dockerArgs),
-        this::validateDockerArgs);
-
-    // not verifying dockerTerminationLogging as it defaults to false
-    // So for
-    //       WorkflowConfiguration.builder()
-    //          .id("styx.TestEndpoint")
-    //          .commitSha(VALID_SHA)
-    //          .schedule(DAYS)
-    //          .serviceAccount("foo@bar.baz.quux")
-    //          .dockerExecConf(new DockerExecConfBuilder()
-    //              .dockerImage("busybox")
-    //              .dockerArgs(List.of("x", "y"))
-    //              .dockerTerminationLogging(true)
-    //              .build())
-    //          .build();
-    // will fail as default at upper level is false but inside dockerExecConf is set to true
+  private void validateConflictingExecConf(List<String> e, WorkflowConfiguration cfg) {
+    var hasOldStyleDockerConf = cfg.dockerImage().isPresent() || cfg.dockerArgs().isPresent();
+    var hasNewStyleDockerConf = cfg.dockerExecConf().isPresent();
+
+    if (hasNewStyleDockerConf && hasOldStyleDockerConf) {
+      e.add("configuration cannot docker parameters in new and old style");
+    }
+    if (cfg.flyteExecConf().isPresent() && (hasNewStyleDockerConf || hasOldStyleDockerConf)) {
+      e.add("configuration cannot specify both docker and flyte parameters");
```

Comment:
shouldn't it be allowed to have flyteExecConf and dockerExecConf ?

Reply:
I guess you are asking because for migrating workflows we said that we will switch workflows from Docker to Flyte. But this PR is not that. My intention with this PR is to have either docker or flyte but not both. Future PRs we will look on how to do the switch and I didn't want to mix concerns

Reply:
right i hear you.
So what am I missing here?
We don't care about the migration/switch here.
So we should have 2 valid cases:
- either it's the old workflow configuration.
- or it's Flyte
It seems to me that the if-statement will trigger whenever Flyte is available.

Reply:
Hm right. So it should be either the FlyteExecConf or the docker configuration. 👍

Reply:
Can we add a comment explaining that we dont need to store docker configuration with flyte, because it's stored on flyte itself?